scripts/gen-iso.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+
+set -uxeE
+
+usage()
+{
+    echo "----------------------------------------------------------------------------------------"
+    echo "- The purpose of this script is to be executed inside the container of our development environment to generate ISO files for XCP-ng."
+    echo "- Usage: $0 <XCP-ng version> <REPOSITORY>"
+    echo "- All options are mandatory."
+    echo "- VERSION of XCP-ng: 8.2 or 8.3"
+    echo "- REPOSITORY: update, testing, ci, ..."
+    echo "- Example: $0 8.3 testing"
+    echo "----------------------------------------------------------------------------------------"
+}
+
+
+VERSION="$1"
+
+if [ $VERSION != "8.3" ] && [ $VERSION != "8.2" ]; then
```

Comment:
variables like `VERSION` which are built from user-provided input should also be protected from special-chars-in-args, using double quotes (paranoid approach is "always quote variable expansion to be sure")

```
[builder@1c3fd59db0dd xcp]$ ./scripts/gen-iso.sh "8.1 != 8.1 -a 8.3" update
+ VERSION='8.1 != 8.1 -a 8.3'
+ '[' 8.1 '!=' 8.1 -a 8.3 '!=' 8.3 ']'
+ REPOSITORY=update
++ date +%Y%m%d
...
+ sudo ./scripts/create-installimg.sh --srcurl http://mirrors.xcp-ng.org/8/8.1 '!=' 8.1 -a 8.3 -o install-8.1 '!=' 8.1 -a 8.3-update.img 8.1 '!=' 8.1 -a 8.3:update
```

Reply:
I don't understand that part.

But I'd like to recall this script isn't meant to be launch directly by an external user. It's mean to be launch by our Jenkins server for internal use. I think we can not insert errors in the arguments.

And the test check the variable. Si if it's something else, it will exit.

Reply:
It just means you should always add `""` around a variable that you test, like in line 20.
```suggestion
if [ "$VERSION" != "8.3" ] && [ "$VERSION" != "8.2" ]; then
```

Be there an actual risk or not in the way the script is used, it's good practice and a good habit to take.